ramlfications/parser/types.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015 Spotify AB
+
+from __future__ import absolute_import, division, print_function
+import attr
+import copy
+from six import MAXSIZE
+from six import iteritems
+
+from ..errors import UnknownTypeError
+from ..parameters import Content
+
+__type_registry = {}
+
+def type_class(klass):
+    if klass.type is not None:
+        __type_registry[klass.type] = klass
+    return klass
+
+def create_type(name, raml_def):
+    # Work on a copy. We will pop the type. Don't modify the raw raml object.
+    raml_def = copy.copy(raml_def)
+
+    # spec:
+    # string is default type when nothing else defined
+    typeexpr = raml_def.pop('type', 'string')
+
+    if typeexpr in __type_registry:
+        return __type_registry[typeexpr](name=name, **raml_def)
+    else:
+        # we start simple, type expressions are for another commit
+        raise UnknownTypeError("{0} type expression is not supported or not defined".format(typeexpr))
+
+@attr.s
+class BaseType(object):
+    name            = attr.ib()
+    description     = attr.ib(default="", repr=False, convert=Content)
```

Comment:
Since the RAML spec says that this will be a markdown string, I would encourage using the same pattern of the rest of the package. I have a [`desc` defined](https://github.com/spotify/ramlfications/blob/master/ramlfications/parameters.py#L86) and then a [`description` property](https://github.com/spotify/ramlfications/blob/master/ramlfications/parameters.py#L104-L108) to handle the parsing of markdown.  It may not be the best approach (open to ideas) but since it's expected to be markdown, we should handle that.